repo/content/content_manager.go

```diff
@@ -755,9 +724,77 @@ func newManagerWithOptions(ctx context.Context, st blob.Storage, f *FormattingOp
 		closed:                make(chan struct{}),
 	}
 
+	if err := setupCaches(ctx, m, caching); err != nil {
+		return nil, errors.Wrap(err, "unable to set up caches")
+	}
+
 	if err := m.CompactIndexes(ctx, autoCompactionOptions); err != nil {
```

Comment:
When we compact index blobs during a maintenance run to remove "old" delete-marked content entries, we intend to iterate through all index blobs and compact them all irrespective of the number of small blobs present. But in getContentsToCompact(), we still rely on logic around the number of very small blobs and might not compact all index blobs. Is that intended behaviour?